cmd/geth/main.go

```diff
@@ -417,6 +428,35 @@ func attach(ctx *cli.Context) {
 	}
 }
 
+func initGenesis(ctx *cli.Context) {
+	genesisPath := ctx.Args().First()
+	if len(genesisPath) == 0 {
+		utils.Fatalf("must supply path to genesis JSON file")
+	}
+
+	var (
+		datadir = utils.MustMakeDataDir(ctx)
+		cache   = ctx.GlobalInt(utils.CacheFlag.Name)
+		handles = utils.MakeDatabaseHandles()
+	)
+
+	chainDb, err := ethdb.NewLDBDatabase(filepath.Join(datadir, "chaindata"), cache, handles)
+	if err != nil {
+		utils.Fatalf("could not open database: %v", err)
+	}
+
+	genesisFile, err := os.Open(genesisPath)
+	if err != nil {
+		utils.Fatalf("failed to read genesis file: %v", err)
+	}
+
+	block, err := core.WriteGenesisBlock(chainDb, genesisFile)
+	if err != nil {
+		utils.Fatalf("failed to write genesis block: %v", err)
+	}
+	glog.V(logger.Info).Infof("successfully wrote genesis block: %x", block.Hash())
```

Comment:
On second thoughts I don't think the log message should be moved at all. Log messages should appear when requested and not forced. I'll add a ` +	glog.V(logger.Info).Infof("successfully wrote genesis block and/or chain rule set: %x", block.Hash())`